### PR TITLE
[Improve][Doris Connector] Unified serialization method,Use RowToJsonConverter and TextSerializationSchema

### PR DESCRIPTION
**Pull request by liugddx**
### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/serialize/SeaTunnelRowSerializer.java [hunk 1]
```diff
@@ -105,46 +79,40 @@ public String parseDeleteSign(RowKind rowKind) {
...
-            return this;
-        }
+        List<String> fieldNames = Arrays.asList(seaTunnelRowType.getFieldNames());
+        List<SeaTunnelDataType<?>> fieldTypes = Arrays.asList(seaTunnelRowType.getFieldTypes());
 
-        public Builder enableDelete(boolean deletable) {
-            this.deletable = deletable;
-            return this;
+        if (enableDelete) {
+            SeaTunnelRow seaTunnelRowEnableDelete = seaTunnelRow.copy();
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/serialize/SeaTunnelRowSerializer.java:100]: reuse `parseDeleteSign`?

**Review by Hisoka-X — COMMENTED** [review]
overall LGTM. Only one minior suggestion.

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/serialize/SeaTunnelRowSerializer.java (hunk 1 above)
Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/serialize/SeaTunnelRowSerializer.java:100]: oh i missed it

**Review by Hisoka-X — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/serialize/SeaTunnelRowConverterTest.java [hunk 2]
```diff
@@ -1,54 +0,0 @@
...
-                "2021-01-01 00:00:00.000001",
-                seaTunnelRowConverter.convert(
-                        LocalTimeType.LOCAL_DATE_TIME_TYPE,
-                        LocalDateTime.of(2021, 1, 1, 0, 0, 0, 1000)));
-        Assertions.assertEquals(
-                "2021-01-01 00:00:00.000123",
-                seaTunnelRowConverter.convert(
-                        LocalTimeType.LOCAL_DATE_TIME_TYPE,
-                        LocalDateTime.of(2021, 1, 1, 0, 0, 0, 123456)));
-    }
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/serialize/SeaTunnelRowConverterTest.java:53]: we should make sure the behavior not change. Please do not delete this test case. Instand of cover it by `SeaTunnelRowSerializer`.

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/serialization/RowBatch.java [hunk 3]
```diff
@@ -71,21 +75,22 @@
 
 @Slf4j
 public class RowBatch {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/serialization/RowBatch.java:77]: why change `RowBatch`? Any test case?

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/serialization/RowBatch.java [hunk 4]
```diff
@@ -267,7 +272,7 @@ private void convertArrowValue(
                         rowIndex ->
                                 float8Vector.isNull(rowIndex) ? null : float8Vector.get(rowIndex));
                 break;
-            case "DECIMAL":
+            case DECIMAL:
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/serialization/RowBatch.java]: why only `DECIMAL` is static field?

File: seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/serialize/SeaTunnelRowConverterTest.java (hunk 2 above)
Comment (liugddx) [reply seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/serialize/SeaTunnelRowConverterTest.java:53]: > we should make sure the behavior not change. Please do not delete this test case. Instand of cover it by `SeaTunnelRowSerializer`.

We use seatunnel's serialization tool, so `SeaTunnelRowConverter.java` was deleted. This test case will no longer be used.

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/serialization/RowBatch.java (hunk 3 above)
Comment (liugddx) [reply seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/serialization/RowBatch.java:77]: > why change `RowBatch`? Any test case?

I adapted the latest version of doris, and I found that many types were incompatible, so I added some type conversions.It is backwards compatible

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/serialization/RowBatch.java (hunk 4 above)
Comment (liugddx) [reply seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/serialization/RowBatch.java]: > why only `DECIMAL` is static field?

My bad, rolled back

**Comment by liugddx** [comment]
After https://github.com/apache/doris/issues/38174 is resolved, I will upgrade to the latest doris version as a image for test cases.

**Comment by Hisoka-X** [comment]
> After [apache/doris#38174](https://github.com/apache/doris/issues/38174) is resolved, I will upgrade to the latest doris version as a image for test cases.

Please create an issue for this.

File: seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/serialize/SeaTunnelRowConverterTest.java (hunk 2 above)
Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/serialize/SeaTunnelRowConverterTest.java:53]: We need all datetime value written to doris to have the same precision.

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/serialization/RowBatch.java [hunk 5]
```diff
@@ -71,21 +75,22 @@
 
 @Slf4j
 public class RowBatch {
+    public static final String DECIMAL = "DECIMAL";
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/serialization/RowBatch.java]:
```suggestion
```

**Review by Hisoka-X — COMMENTED** [review]
LGTM except one minor problem.

**Review by Hisoka-X — APPROVED** [review]

**Review by wuchunfu — APPROVED** [review]

**Pull request merged**